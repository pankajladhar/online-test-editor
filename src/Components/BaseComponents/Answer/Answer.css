.Answer {
  cursor: pointer;
  padding: 10px 15px;
  border: 1px transparent solid;
  display: flex;
  align-content: center;
  margin-bottom: 20px; }
  @media (max-width: 760px) {
    .Answer {
      margin-bottom: 0; } }
  .Answer:hover {
    border: 1px #fff solid; }
  .Answer.checked {
    border: 1px #349284 solid; }
    .Answer.checked:hover {
      border: 1px #349284 solid; }
  .Answer .RadioButton {
    margin-left: 15px;
    width: 100%; }
    .Answer .RadioButton input[type="radio"] {
      display: none; }
    .Answer .RadioButton label {
      margin: 0;
      display: flex;
      cursor: pointer; }
