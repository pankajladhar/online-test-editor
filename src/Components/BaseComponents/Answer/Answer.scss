.Answer {  
  cursor: pointer;
  padding: 10px 15px;
  border: 1px transparent solid;
  display: flex;
  align-content: center;
  margin-bottom: 20px;

  &:hover {
    border: 1px #fff solid;
  }

  &.checked {
    border: 1px #349284 solid;
    &:hover {
      border: 1px #349284 solid;
    }
  }

  .RadioButton {
    margin-left: 15px;
    width: 100%;
    input[type="radio"] {
      display: none;
    }
    label {
      margin: 0;
      display: flex;
      cursor: pointer;;
    }
  }
}