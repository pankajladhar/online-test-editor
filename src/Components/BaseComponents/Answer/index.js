import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RadioButton from './../RadioButton';
import './Answer.css';

class Answer extends Component {
    constructor(props) {
        super(props);  
    }
    
    render() {
        return (
            <div className={`Answer ${this.props.checked ? 'checked' : 'unchecked'}`}>
                <span>{this.props.index + 1}.</span>
                <RadioButton
                    checked={this.props.checked}
                    id={this.props.id}
                    onChange={this.props.onChange}
                    data={this.props.data} />
            </div>
        );
    }
}

Answer.propTypes = {

};

export default Answer;