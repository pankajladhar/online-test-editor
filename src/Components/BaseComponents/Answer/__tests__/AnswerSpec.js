import React from 'react';
import renderer from 'react-test-renderer';
import Answer from './../index';

describe('Answer', () => {
    let answerProps = {}
    beforeEach(() => {
        answerProps = {
            index: 1,
            id: "Answer",
            onChange: jest.fn(),
            data: "some string"
        }
    });

    it('renders correctly when checked is true', () => {
        answerProps.checked = true;
        let tree = renderer.create(
            <Answer {...answerProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when checked is false', () => {
        answerProps.checked = false;
        let tree = renderer.create(
            <Answer {...answerProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});