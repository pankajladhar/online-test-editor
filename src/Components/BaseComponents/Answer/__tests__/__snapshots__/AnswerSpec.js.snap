// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer renders correctly when checked is false 1`] = `
<div
  className="Answer unchecked"
>
  <span>
    2
    .
  </span>
  <div
    className="RadioButton"
  >
    <input
      checked={false}
      id="Answer"
      name="answers"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="Label "
      htmlFor="Answer"
    >
      some string
    </label>
  </div>
</div>
`;

exports[`Answer renders correctly when checked is true 1`] = `
<div
  className="Answer checked"
>
  <span>
    2
    .
  </span>
  <div
    className="RadioButton"
  >
    <input
      checked={true}
      id="Answer"
      name="answers"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="Label "
      htmlFor="Answer"
    >
      some string
    </label>
  </div>
</div>
`;
